src/Analyzer.TemplateProcessor.UnitTests/ArmTemplateProcessorTests.cs

```diff
@@ -665,6 +665,82 @@ public void CopyResourceDependants_DependsOnIsSubResource_CopiesToCorrectResourc
             AssertDictionariesAreEqual(expectedMapping, armTemplateProcessor.ResourceMappings);
         }
 
+        [TestMethod]
+        public void CopyResourceDependants_DependsOnSpecifiedInNestedChildResource()
+        {
+            // Arrange
+            string resourceJson = @"{
+                ""type"": ""Microsoft.Sql/servers"",
+                ""apiVersion"": ""2020-02-02-preview"",
+                ""name"": ""sqlServerName"",
+                ""location"": ""location"",
+                ""properties"": {
```

Comment:
Optional: To reduce the number of lines in this test, you can make all the `properties` sections empty objects. They're not important for this test.